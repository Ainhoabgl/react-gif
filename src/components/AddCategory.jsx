// Este componente contiene un input donde el usuario puede escribir una categoría nueva.
//Cuando el usuario presiona "Enter", la nueva categoría se envía a GifExpertApp.

import { useState } from "react"
import { PropTypes } from "prop-types";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('')
    const onInputChange= ({target}) => {
        setinputValue(target.value);
    }

    const onSubmit = (event)=> {
        event.preventDefault();
        //si la letra es menor a 1 no se añadirá a la lista
        if(inputValue.trim().length <= 1)return;
        // setCategorias(categorias=>[inputValue, ...categorias])
        //cada vez que se envíe una petición se limpiará el contenido del input
        setinputValue('');
        onNewCategory(inputValue.trim());
    }
  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input 
            type="text" 
            placeholder="Buscar Gifs"
            value={inputValue}
            onChange={onInputChange}
        />
    </form>
    
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
