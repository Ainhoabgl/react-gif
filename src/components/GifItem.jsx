import PropTypes from 'prop-types';

// Este componente muestra un solo GIF dentro de una tarjeta (<div className="card">).
export const GifItem = ({title, url}) => {

    return (
    <div className="card">

        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

/* Tarea

1. AÃ±adir PropTypes
  a. title obligatorio
  b. url obligatorio

2. Evaluar el sanpshot

*/

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired

}