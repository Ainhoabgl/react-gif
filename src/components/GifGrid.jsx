// Este componente recibe una categoría y se encarga de:

// Hacer la petición a la API de Giphy (a través de getGifs).

// Guardar las imágenes en el estado (images).

// Renderizar una lista de <GifItem /> con los GIFs obtenidos.

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ categoria }) => {
    const { images, isLoading } = useFetchGifs(categoria);

    return (
        <>
            <h3>{categoria}</h3>
            {/* si isLoading esta en tru pondra el cargando si no nada */}
            {isLoading && <h2>Cargando...</h2>} {/* Muestra un mensaje mientras carga */}

            <div className="card-grid">
                {images.map((image) => (
                    <GifItem 
                        key={image.id}
                        {...image}
                    />
                ))}
            </div>
        </>
    );
};

