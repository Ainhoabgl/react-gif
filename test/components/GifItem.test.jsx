import { render, screen } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem";

//Este c√≥digo prueba que el componente se renderiza correctamente usando un "snapshot"
describe('Pruebas en <GifItem/>', () => { 
    const title="Goku" 
    const url="https://goku.com/gif.gif"

    // Primer test: Comprueba que el componente coincide con el snapshot.
    test('Debe de coincidir con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
     })

    //Segundo test: Verifica que la imagen tiene el src y alt correctos.
     test ('debe de mostrar la imagen con el URL Y el ALT indicado', ()=> {
        render (<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src)
        // console.log(screen.getByRole('img').src).toBe(url);

        const {src, alt}= screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

     })
     //Tercer test: Verifica que la imagen tenga un titulo
     test ('debe de mostrar el titulo en el componente', ()=> {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
     })

 })