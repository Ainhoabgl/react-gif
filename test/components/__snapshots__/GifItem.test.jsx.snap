// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.com/gif.gif"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
